other/check-vpa-configuration/check-vpa-configuration.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-matching-vpa
+  annotations:
+    policies.kyverno.io/title: Ensure Matching VerticalPodAutoscaler 
+    policies.kyverno.io/category: Availability
+    policies.kyverno.io/severity: medium
+    kyverno.io/kyverno-version: 1.11.4 
+    kyverno.io/kubernetes-version: "1.27"     
+    policies.kyverno.io/subject: Deployment, StatefulSet, ReplicaSet, DaemonSet
+    policies.kyverno.io/description: >-
+      This policy enforces that Deployments, StatefulSets, Replicasets, and DaemonSets have 
+      a corresponding VerticalPodAutoscaler to ensure potential automatic resource adjustments. 
+spec:
+  validationFailureAction: Audit
+  background: false
+  rules:
+    - name: require-vpa
+      match:
+        any:
+        - resources:
+            kinds:
+              - Deployment
+              - StatefulSet
+              - ReplicaSet
+              - DaemonSet
+      validate:
+        message: >-
+          Workload '{{request.object.kind}}/{{request.object.metadata.name}}' 
+          requires a matching VerticalPodAutoscaler (VPA) in the 
+          '{{request.object.metadata.namespace}}' namespace.
+      context:
+      - name: metricsServer
+        apiCall:
+          urlPath: "/apis/metrics.k8s.io/v1beta1/namespaces/{{ request.namespace }}/pods"
+          jmesPath:  
+      preconditions:
+        any:
+        - key: "{{ metricsServer }}"
+          operator: GreaterThan
+          value: 0 
```

Comment:
I don't understand why you're now checking for metrics server. This doesn't have anything to do with whether a Deployment/StatefulSet/ReplicaSet/DaemonSet has a matching VPA.

Reply:
actually it a requirement of the vpa to have a metrics server but I think I can handle this in a separate issue

Reply:
But that isn't what this policy is intending to do. This policy, at least what it appears so far, is supposed to check new Deployments etc to ensure they have a VPA which already exists. You just need to check all VPAs in the destination Namespace in their `spec.targetRef` section. It has nothing to do with metrics server.